components/suggest/README.md

```diff
@@ -1,7 +1,63 @@
+
+**3. Use the `query()` and `interrupt()` methods to query for fresh suggestions as the user types.** The application passes the user's input, and additional options like which suggestion types to return, to `query()`. Querying the database is disk I/O-bound, so it's very important for each application to use its platform's facilities for asynchronous work to avoid blocking its main thread. For example, Firefox Android uses Kotlin coroutines, while iOS uses Swift actors. (The UniFFI bindings for Firefox Desktop take care of this automatically). Running `query()` off-main-thread also lets applications `interrupt()` those queries when the user changes their input. This avoids waiting for a query to return suggestions that are now stale.
+
+### For contributors
+
+This section is a primer for engineers contributing code to the Suggest Rust component.
+
+`suggest.udl` describes the component's interface for foreign language consumers. UniFFI uses this file to generate the language bindings for each platform. If you're adding a new suggestion type, you'll want to update the declarations of `Suggestion` and `SuggestionProvider` in this file, as well as the definitions of those types in `suggestion.rs` and `provider.rs`, respectively.
+
+`store.rs` contains the implementation of `SuggestStore` and most of the component's tests.
+
+`schema.rs` manages the database schema. **Remember to bump the schema version and add a migration whenever you change the schema.**
+
+`db.rs` interacts with the Suggest database. The `SuggestDao` type in this module is a ["data access object"](https://en.wikipedia.org/wiki/Data_access_object) (DAO) that contains all the SQL statements for querying and updating the SQLite database. By convention, `SuggestDao` methods that can write to the database take `&mut self`, and methods that only read take `&self`. The `SuggestDb::read()` and `SuggestDb::write()` methods take a closure that receives either `&SuggestDao` or `&mut SuggestDao`; this is how we enforce that all writes are done in a transaction. If you're curious to learn about how we use SQLite, diagnosing a slow query, or adding a new suggestion type, you'll almost certainly want to look at this module.
+
+`rs.rs` defines all the Remote Settings [record](https://docs.kinto-storage.org/en/stable/concepts.html#buckets-collections-and-records) and [attachment](https://docs.kinto-storage.org/en/stable/faq.html#can-i-store-files-inside-kinto) types. The records in the `quicksuggest` Remote Settings collection don't store the suggestions themselves. Instead, each record has a type, and a pointer to a JSON attachment that contains multiple suggestions of that type. This module defines [Serde](https://serde.rs/)-compatible types for these records and attachments.
```

Comment:
There is also a [hidden gem](https://github.com/mozilla/application-services/blob/main/components/suggest/src/rs.rs#L6-L32) documented in rs.rs, shall we add a pointer to that here as well?